catalog/spatial/geowebcache/geowebcache-admin-plugin/src/main/webapp/js/model/GwcLayers.js

```diff
@@ -0,0 +1,163 @@
+                var that = this;
+                $.ajax({
+                    url: url,
+                    type : 'DELETE',
+                    success: function() {
+                        var layers = that.get("layers");
+                        $.each(layers, function() {
+                            if(name === this.name) {
+                                layers.splice( $.inArray(this, layers), 1 );
+                                that.set("layers", layers);
+                                that.trigger('change:layers', this);
+                            }
+                        });
+                    }
+                });
+            },
+            postLayerData: function(layerData, type) {
+                var restUrl = "../../geowebcache/rest/layers/" + layerData.name + ".xml";
+                var that = this;
+                var data = this.generateXmlForLayer(layerData);
+                $.ajax({
+                    url: restUrl,
+                    type : type,
+                    dataType : "xml",
+                    data : data,
+                    success: function() {
+                        that.getLayers();
+                    }
+                });
+            },
+            addLayer: function(layerData) {
+                this.postLayerData(layerData, "PUT");
+            },
+            updateLayer: function(layerData) {
+                if(this.nameExistsInArray(layerData.name)) {
+                    this.postLayerData(layerData, "POST");
+                } else {
+                    this.deleteLayer(layerData.oldName);
+                    this.postLayerData(layerData, "PUT");
+                }
+            },
+            nameExistsInArray: function(name) {
+                var exists = false;
+                $.each(this.get("layers"), function(index, value) {
+                    if(name === value.name) {
+                        exists = true;
+                    }
+                });
+                return exists;
+            },
+            getLayer: function(name) {
+                var layer;
+                $.each(this.get("layers"), function(index, value) {
+                    if(name === value.name) {
+                        layer = value;
+                    }
+                });
+                return layer;
+            },
+            generateXmlForLayer: function(layerData) {
+                var data;
+                data = "<wmsLayer>";
+                data+= "    <name>" + layerData.name + "</name>";
+                data+= "    <mimeFormats>";
+                $.each(layerData.mimes, function(index,value) {
+                    data +=  "      <string>" + value + "</string>";
+                });
+                data+= "    </mimeFormats>";
+                data+= "    <wmsUrl>";
+                data+= "        <string>" + layerData.url + "</string>";
+                data+= "    </wmsUrl>";
```

Comment:
This is certainly one way to generate the XML. Does JavaScript have format strings? That might be a cleaner approach. Then the XML can be pulled out as a resource, to be used as a template.